packages/date-time/src/components/CalendarDayGrid/CalendarDayGrid.base.tsx

```diff
@@ -778,4 +764,27 @@ export class CalendarDayGridBase extends BaseComponent<ICalendarDayGridProps, IC
       });
     };
   };
+
+  /**
+   * When given work week, if the days are non-contiguous, the hover states look really weird. So for non-contiguous work weeks,
+   * we'll just show week view instead
+   */
+  private getDateRangeTypeToUse = (dateRangeType: DateRangeType, workWeekDays: DayOfWeek[] | undefined): DateRangeType => {
+    if (workWeekDays && dateRangeType === DateRangeType.WorkWeek) {
+      const sortedWWDays = workWeekDays.sort();
+      let isContiguous = true;
+      for (let i = 1; i < sortedWWDays.length; i++) {
+        if (sortedWWDays[i] !== sortedWWDays[i - 1] + 1) {
+          isContiguous = false;
+          break;
+        }
+      }
```

Comment:
If someone passes an empty array for `workWeekDays`, will this yield the expected result? (Not sure what would be preferred, but I believe it would return `dateRangeType` which might be confusing.)

Reply:
good point, if they have no work days we probably want DateRangeType.Week